code/modules/SCP/scp017.dm

```diff
@@ -0,0 +1,637 @@
+			light_sources_in_range += M
+
+	// Movement based on AI state
+	switch(current_ai_state)
+		if(IDLE)
+			// Stay still or wander slowly in shadows
+			if(Tturf.get_lumcount() > shadow_threshold)
+				// If in light, try to find a shadow
+				var/turf/target_turf = find_darkest_adjacent_turf()
+				if(target_turf)
+					step_to(src, target_turf)
+			else
+				// Wander slowly
+				if(prob(10)) // Small chance to wander
+					step(src, pick(NORTH, SOUTH, EAST, WEST))
+
+		if(HUNTING)
+			var/mob/living/target_mob = get_highest_priority_target()
+			if(target_mob)
+				if(light_sources_in_range.len > 0)
+					// Fleeing light sources
+					var/turf/flee_turf = find_flee_turf(light_sources_in_range)
+					if(flee_turf)
+						Move(flee_turf, ignore_light_check = TRUE)
+					else
+						// If no clear flee path, try to find darkest adjacent
+						var/turf/dark_turf = find_darkest_adjacent_turf()
+						if(dark_turf)
+							step_to(src, dark_turf) // This step is still to a dark turf, so no ignore_light_check needed
+				else
+					// Move towards target, prioritizing shadows
+					var/turf/target_turf = get_turf(target_mob)
+					if(target_turf)
+						Goto(target_turf)
+
+		if(STALKING)
+			var/mob/living/target_mob = get_highest_priority_target()
+			if(target_mob)
+				// Move slower and deliberately, prioritize shadows
+				turns_per_move = 2 // Slower movement
+				var/turf/target_turf = get_turf(target_mob)
+				if(target_turf)
+					Goto(target_turf)
+			else
+				turns_per_move = initial(turns_per_move) // Reset to normal
+				current_ai_state = IDLE // No target, go back to idle
+
+		if(VINDICTIVE)
+			// Prioritize breaking lights or attacking specific targets
+			var/obj/machinery/light/light_to_break
+			for(var/obj/machinery/light/L in range(light_suppression_range, src))
+				if(L.on)
+					light_to_break = L
+					break
+			if(light_to_break)
+				step_to(src, light_to_break)
+				if(get_dist(src, light_to_break) <= 1)
+					UnarmedAttack(light_to_break) // Directly attack the light
+			else
+				var/mob/living/target_mob = get_highest_priority_target()
+				if(target_mob)
+					step_to(src, target_mob)
+
+/mob/living/simple_animal/hostile/scp017/proc/find_darkest_adjacent_turf()
+	var/turf/darkest_turf
+	var/min_lumcount = INFINITY
+	for(var/dir in GLOB.alldirs) // Check all 8 directions: cardinals and diagonals - Substitute for .Adjacent() not working here.
+
+		var/turf/T = get_step(get_turf(src), dir)
+		var/area/A = get_area(T)
+
+		if(A.area_lighting == AREA_LIGHTING_DYNAMIC)
+			var/lum = T.get_lumcount()
+			if(lum < min_lumcount)
+				min_lumcount = lum
+				darkest_turf = T
+	if(darkest_turf && darkest_turf.get_lumcount() <= shadow_threshold)
+		return darkest_turf
+	return null
+
+/mob/living/simple_animal/hostile/scp017/proc/find_flee_turf(list/light_sources)
+	var/turf/current_turf = get_turf(src)
+	var/turf/flee_turf
+	var/max_distance_from_lights = -1
+
+	for(var/turf/T in current_turf.Adjacent())
```

Comment:
Adjacent() is for checking adjacency. Maybe you want `get_adjacent_open_turfs(current_turf)`?

Reply:
Oh wow I completely missed the mark here.